package.json

```diff
@@ -52,6 +52,89 @@
     "onCommand:java.formatterSettings"
   ],
   "contributes": {
+    "walkthroughs": [
+      {
+        "id": "javaWelcome",
+        "title": "Dive into Java Development",
+        "description": "Your first steps to set up powerful Java tools in a lightweight, performant editor!",
+        "tasks": [
+          {
+            "id": "java.runtime",
+            "title": "Get your Java runtime ready",
+            "description": "The built-in configurator makes it easy to set up your preferred Java runtime.\n[Configure Java Runtime](command:toSide:java.runtime)",
+            "media": {
+              "path": "walkthrough/runtime.png",
+              "altText": "Configure Java Runtime with tabs for Project SDKs"
+            }
+          },
+          {
+            "id": "java.extGuide",
+            "title": "Support additional tools & frameworks",
+            "description": "Got additional support for major tool chains, frameworks and application servers.\n[Recommended Java Extensions](command:toSide:java.extGuide)",
+            "media": {
+              "path": "walkthrough/extGuide.png",
+              "altText": "Open Project Folder"
+            }
+          },
+          {
+            "id": "java.openProjectFolder",
+            "title": "Open project folder",
+            "description": "Open an existing folder that contains your Java project, or create a new folder.\n[Open Folder...](command:workbench.action.files.openFolder)\n[Create new Project](command:java.project.create)",
+            "media": {
+              "path": "walkthrough/open-project.png",
+              "altText": "Open Project Folder"
+            },
+            "when": "!isMac && workspaceFolderCount == 0"
+          },
+          {
+            "id": "java.openProjectFolder.mac",
+            "title": "Open project folder",
+            "description": "Open an existing folder that contains your Java project, or create a new folder.\n[Open Folder...](command:workbench.action.files.openFileFolder)\n[Create new Project](command:java.project.create)",
+            "media": {
+              "path": "walkthrough/open-project.png",
+              "altText": "Open Project Folder"
+            },
+            "when": "isMac && workspaceFolderCount == 0"
+          },
+          {
+            "id": "java.showProjectExplorer",
+            "title": "Explore your project",
+            "description": "View class paths and dependencies, and create new projects, packages, and classes.\n[Show Java Projects](command:javaProjectExplorer.focus)",
```

Comment:
How to handle the case that the extension is not activated? (command `javaProjectExplorer.focus` not exists)

Adding a when clause?